_pages/documentation/CPU-Models/TraceCPU.md

```diff
@@ -0,0 +1,159 @@
+
+ **Publication**:
+
+[Exploring System Performance using Elastic Traces: Fast, Accurate and Portable"](https://ieeexplore.ieee.org/document/7818336) Radhika Jagtap, Stephan Diestelhorst, Andreas Hansson, Matthias Jung and Norbert Wehn SAMOS 2016
+
+**Trace generation and replay methodology**
+
+![Methodology block diagram showing elastic trace generation using O3 CPU and replay using Trace CPU
+](../../../assets/img/Etrace_methodology.jpg) 
+
+## **Elastic Trace Generation**
+The Elastic Trace Probe Listener listens to Probe Points inserted in O3 CPU pipeline stages. It monitors each instruction and creates a dependency graph by recording data Read-After-Write dependencies and order dependencies between loads and stores. It writes the instruction fetch request trace and the elastic data memory request trace as two separate files as shown below.
+
+![Elastic trace file generation](../../../assets/img/Etraces_output.jpg)
+
+### **Trace file formats**
+
+The elastic data memory trace and fetch request trace are both encoded using google protobuf.
+
+##### **Elastic Trace fields in protobuf format**
+
+Fields    | Discritption
+-------------- | -------------
+required uint64 seq_num &nbsp;   | Instruction number used as an id for tracking dependencies
+required RecordType type &nbsp;    | RecordType enum has values: INVALID, LOAD, STORE, COMP
+optional uint64 p_addr &nbsp; 	| Physical memory address if instruction is a load/store
+optional uint32 size &nbsp; 	| Size in bytes of data if instruction is a load/store
+optional uint32 flags &nbsp; 	| 	Flags or attributes of the access, ex. Uncacheable
+required uint64 rob_dep &nbsp;  |   Past instruction number on which there is order (ROB) dependency
+required uint64 comp_delay &nbsp;       |	Execution delay between the completion of the last dependency and the execution of the instruction &nbsp;
+repeated uint64 reg_dep &nbsp;              | Past instruction number on which there is RAW data dependency
+optional uint32 weight &nbsp; | 	To account for committed instructions that were filtered out
+optional uint64 pc &nbsp; | Instruction address, i.e. the program counter
+optional uint64 v_addr &nbsp; | 	Virtual memory address if instruction is a load/store
+optional uint32 asid &nbsp; | Address Space ID
+
+A decode script in Python is available at `util/decode_inst_dep_trace.py` that outputs the trace in ASCII format.
+
+**Example of a trace in ASCII**
+
+1,356521,COMP,8500::
```

Comment:
You might want to make this a verbatim block by inserting a tab at the beginning of each line. Alternatively, you could wrap in ```

Reply:
Fixed it in new commit :)